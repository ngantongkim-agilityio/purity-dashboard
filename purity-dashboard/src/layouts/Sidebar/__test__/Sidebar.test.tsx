import { render } from '@testing-library/react';

// Components
import { Sidebar } from '..';

jest.mock('next-auth', () => ({
  __esModule: true,
  default: jest.fn().mockReturnValue({
    handlers: jest.fn(),
    signOut: jest.fn(),
    auth: jest.fn()
  })
}));

jest.mock('@/services', () => ({
  getUser: jest.fn()
}));

describe('Sidebar component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should render Sidebar component with default props', () => {
    const { container } = render(<Sidebar />);

    expect(container).toMatchSnapshot();
  });
});
