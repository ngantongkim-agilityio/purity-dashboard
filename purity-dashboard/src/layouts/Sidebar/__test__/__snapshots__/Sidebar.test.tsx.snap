// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component Should render Sidebar component with default props 1`] = `
<div>
  <aside
    class="flex h-full flex-col p-8 min-w-[310px] space-y-6"
  >
    <a
      class="flex justify-start space-x-3 px-4 py-2"
      href="/"
    >
      <svg
        fill="none"
        height="22"
        width="22"
      >
        <path
          d="M20.88 0H1.122C.502 0 0 .502 0 1.122v9.408c0 .62.502 1.122 1.122 1.122h4.054v9.228A1.12 1.12 0 0 0 6.296 22h9.408c.62 0 1.122-.502 1.122-1.122v-9.223h4.053A1.12 1.12 0 0 0 22 10.534V1.122A1.12 1.12 0 0 0 20.88 0ZM1.3 1.302h9.05v9.05H1.3v-9.05ZM15.525 20.7H6.476v-9.045h9.05V20.7Zm5.175-10.35h-9.05v-9.05h9.05v9.05Z"
          fill="#2D3748"
        />
        <path
          d="M7.448 7.325a.65.65 0 0 0 .92-.92L6.291 4.327a.654.654 0 0 0-.92 0L3.288 6.405a.65.65 0 1 0 .92.92l1.617-1.618 1.623 1.618ZM14.557 7.325l1.618-1.618 1.617 1.618a.65.65 0 0 0 .92-.92l-2.077-2.078a.654.654 0 0 0-.92 0l-2.078 2.078a.65.65 0 0 0 .92.92ZM9.382 14.677a.65.65 0 0 0-.92.921l2.079 2.076a.65.65 0 0 0 .92 0l2.077-2.076a.651.651 0 1 0-.92-.921l-1.617 1.619-1.62-1.62Z"
          fill="#2D3748"
        />
      </svg>
      <h1
        class="text-secondary-300 text-sm uppercase font-lora font-medium"
      >
        Purity UI Dashboard
      </h1>
    </a>
    <hr
      class="shrink-0 bg-divider border-none w-full h-divider bg-gradient-to-r from-grey-100 from-0% via-transparent to-grey-100 to-100%"
      role="separator"
      variant="linear"
    />
    <div
      class="flex grow flex-row justify-between space-x-4 md:flex-col md:space-x-0 md:space-y-2"
    >
      <nav
        class="flex grow flex-row md:grow-0 justify-between md:flex-col space-x-4 md:space-x-0 md:space-y-2 sm:mr-16 md:mr-0"
      >
        <a
          class="flex grow justify-center items-center gap-3 rounded-xl bg-secondary-100 text-secondary-200 text-xs font-bold hover:bg-secondary-50 md:flex-none md:justify-start p-3"
          href="/dashboard"
        >
          <div
            class="p-2 bg-primary-100 text-primary rounded-xl md:shadow-secondary-50 md:shadow-md"
          >
            <svg
              class="w-4 h-4"
              fill="none"
            >
              <g
                clip-path="url(#a)"
                fill="currentColor"
              >
                <path
                  d="M7.663 3.468a.234.234 0 0 0-.324 0L1.945 8.62a.234.234 0 0 0-.072.17l-.001 4.835a.938.938 0 0 0 .937.938h2.816a.469.469 0 0 0 .469-.469V10.11a.234.234 0 0 1 .234-.235h2.344a.234.234 0 0 1 .234.235v3.984a.469.469 0 0 0 .469.469h2.814a.937.937 0 0 0 .938-.938V8.79a.232.232 0 0 0-.073-.17L7.663 3.468Z"
                />
                <path
                  d="m14.382 7.654-2.191-2.097V2.376a.469.469 0 0 0-.469-.469h-1.406a.469.469 0 0 0-.469.469v.937L8.15 1.691a.912.912 0 0 0-.65-.253.91.91 0 0 0-.648.253L.62 7.653a.478.478 0 0 0-.04.656.469.469 0 0 0 .68.03l6.08-5.808a.234.234 0 0 1 .323 0l6.08 5.809a.469.469 0 0 0 .662-.013.48.48 0 0 0-.023-.673Z"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 .5h15v15H0z"
                    fill="currentColor"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="hidden md:block"
          >
            Dashboard
          </p>
        </a>
        <a
          class="flex grow justify-center items-center gap-3 rounded-xl bg-secondary-100 text-secondary-200 text-xs font-bold hover:bg-secondary-50 md:flex-none md:justify-start p-3"
          href="/dashboard/authors"
        >
          <div
            class="p-2 bg-primary-100 text-primary rounded-xl md:shadow-secondary-50 md:shadow-md"
          >
            <svg
              class="w-4 h-4"
              fill="none"
            >
              <g
                clip-path="url(#a)"
                fill="currentColor"
              >
                <path
                  d="M3.047 15.032h-.938a.703.703 0 0 1-.703-.704V10.11a.703.703 0 0 1 .703-.703h.938a.703.703 0 0 1 .703.703v4.219a.703.703 0 0 1-.703.703ZM9.61 15.03h-.938a.703.703 0 0 1-.703-.702V7.296a.703.703 0 0 1 .703-.703h.937a.703.703 0 0 1 .703.703v7.032a.703.703 0 0 1-.703.703ZM12.89 15.031h-.937a.703.703 0 0 1-.703-.703V4.015a.703.703 0 0 1 .703-.703h.938a.703.703 0 0 1 .703.704v10.312a.703.703 0 0 1-.703.703ZM6.328 15.031h-.937a.703.703 0 0 1-.704-.703V1.672a.703.703 0 0 1 .704-.703h.937a.703.703 0 0 1 .703.703v12.656a.703.703 0 0 1-.703.703Z"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 .5h15v15H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="hidden md:block"
          >
            Authors
          </p>
        </a>
        <a
          class="flex grow justify-center items-center gap-3 rounded-xl bg-secondary-100 text-secondary-200 text-xs font-bold hover:bg-secondary-50 md:flex-none md:justify-start p-3"
          href="/dashboard/products"
        >
          <div
            class="p-2 bg-primary-100 text-primary rounded-xl md:shadow-secondary-50 md:shadow-md"
          >
            <svg
              class="w-4 h-4"
              fill="none"
            >
              <g
                clip-path="url(#a)"
                fill="currentColor"
              >
                <path
                  d="M.937 11.515a1.64 1.64 0 0 0 1.641 1.64h9.844a1.641 1.641 0 0 0 1.64-1.64V7.004H.937v4.511ZM2.871 9.29a.879.879 0 0 1 .879-.88h1.406a.879.879 0 0 1 .88.88v.586a.879.879 0 0 1-.88.879H3.75a.879.879 0 0 1-.879-.88V9.29ZM12.422 2.843H2.578a1.64 1.64 0 0 0-1.64 1.64v.762h13.124v-.761a1.64 1.64 0 0 0-1.64-1.641Z"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 .5h15v15H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="hidden md:block"
          >
            Products
          </p>
        </a>
        <a
          class="flex grow justify-center items-center gap-3 rounded-xl bg-secondary-100 text-secondary-200 text-xs font-bold hover:bg-secondary-50 md:flex-none md:justify-start p-3"
          href="/dashboard/rtl"
        >
          <div
            class="p-2 bg-primary-100 text-primary rounded-xl md:shadow-secondary-50 md:shadow-md"
          >
            <svg
              class="w-4 h-4"
              fill="none"
            >
              <g
                clip-path="url(#a)"
              >
                <path
                  d="M13.757 4.031a.468.468 0 0 0-.749-.117l-1.8 1.801a.472.472 0 0 1-.663 0l-.778-.779a.469.469 0 0 1 0-.663L11.56 2.48a.468.468 0 0 0-.14-.759c-1.354-.605-3.044-.29-4.11.768-.905.9-1.175 2.304-.74 3.854a.466.466 0 0 1-.13.468l-4.878 4.455a1.88 1.88 0 1 0 2.654 2.655l4.502-4.888A.466.466 0 0 1 9.18 8.9c.442.121.897.184 1.355.187.978 0 1.837-.317 2.456-.927 1.147-1.13 1.322-3.047.766-4.128Zm-10.842 9.57a.937.937 0 1 1-.203-1.864.937.937 0 0 1 .203 1.864Z"
                  fill="currentColor"
                />
              </g>
              <defs>
                <clippath
                  id="a"
                >
                  <path
                    d="M0 .5h15v15H0z"
                    fill="#fff"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <p
            class="hidden md:block"
          >
            RTL
          </p>
        </a>
      </nav>
      <div
        class="hidden h-auto grow md:block"
      >
        <div
          class="relative rounded-xl w-full h-[169px] mt-16"
        >
          <img
            alt="Need help background"
            class="transition-opacity object-cover duration-200 ease-in-out opacity-80 rounded-xl"
            data-nimg="fill"
            decoding="async"
            sizes="100vw"
            src="http://localhost/_next/image?url=%2Fimages%2Fhelp-background.webp&w=3840&q=75"
            srcset="/_next/image?url=%2Fimages%2Fhelp-background.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fhelp-background.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fhelp-background.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fhelp-background.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fhelp-background.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fhelp-background.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fhelp-background.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fhelp-background.webp&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          />
          <div
            class="absolute top-0 left-0 z-10 p-3 h-full w-full"
          >
            <div
              class="bg-primary-100 p-2 w-fit rounded-xl mb-6"
            >
              <svg
                class="w-[18px] h-[18px] text-primary"
                fill="none"
              >
                <path
                  d="M9 0C4.03 0 0 4.031 0 9s4.031 9 9 9 9-4.031 9-9-4.031-9-9-9Zm-.281 14.25a.938.938 0 1 1 0-1.875.938.938 0 0 1 0 1.875Zm1.567-4.781c-.76.51-.864.977-.864 1.406a.656.656 0 1 1-1.313 0c0-1.027.473-1.844 1.445-2.497.904-.606 1.415-.99 1.415-1.836 0-.574-.329-1.01-1.008-1.334-.16-.076-.515-.15-.953-.145-.55.007-.976.139-1.305.403-.62.499-.672 1.042-.672 1.05a.657.657 0 0 1-1.312-.064c.005-.114.084-1.14 1.16-2.006.559-.449 1.27-.682 2.11-.693.595-.007 1.155.094 1.534.273 1.135.537 1.758 1.432 1.758 2.516 0 1.586-1.06 2.298-1.995 2.927Z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <h3
              class="text-primary-100 text-sm font-bold"
            >
              Need help?
            </h3>
            <p
              class="text-primary-100 font-normal text-xs"
            >
              Please check our docs
            </p>
            <button
              class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover border-transparent font-semibold p bg-primary-100 uppercase text-grey-300 px-4 min-w-20 h-9 text-xs gap-2 rounded-xl mt-2 w-full"
              type="button"
            >
              Documentation
            </button>
          </div>
        </div>
      </div>
      <div>
        <button
          class="flex h-[48px] w-full grow items-center justify-center gap-2 rounded-md bg-gray-50 p-3 text-sm font-medium hover:bg-sky-100 hover:text-blue-600 md:flex-none md:justify-start md:p-2 md:px-3"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
          >
            <path
              d="M5.636 5.636a9 9 0 1 0 12.728 0M12 3v9"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <div
            class="hidden md:block"
          >
            Sign Out
          </div>
        </button>
      </div>
    </div>
  </aside>
</div>
`;
